### PR TITLE
[ARCTIC-389] Automatic toggle switch to enable double write

### LINKED ISSUE #389: [Feature]: Automatic toggle switch to enable double write
### Description

As of now, we have to manually config the `arctic.emit.mode` to determine which stores the flink writers should emit data to before the job launch. There are three enums of the `arctic.emit.mode`: 'file', 'log', 'file,log'.

### Use case/motivation

Use case:
All changelog and historical data ingress into the arctic stores. Writing historical data into filestore and writing incremental data into logstore is expected.

### Describe the solution

Add a new enum to the `arctic.emit.mode`: 'auto'. It means that the flink job writes data into the filestore first, then writes data into the logstore and the filestore immediately after matching an appropriate pattern.
```
create source(
tm timestamp(3),
WATERMARK FOR tm AS tm ) with (...);

insert into arctic /* +OPTIONS ('arctic.emit.auto-write-to-logstore.watermark-gap'='10s')*/
from source;
```

| Config     | default  | type | Note                                     |
| --------- |------------|------------|------------------------------------------|
| arctic.emit.mode     | auto |String | file, log, auto. e.g. <br> 'file' means only writing data into filestore.  <br> 'log' means only writing data into logstore.  <br> 'file,log' means writing data into both filestore and logstore. <br> 'auto' means writing data into filestore if the logstore of the arctic table is disabled; Also means writing data into both filestore and logstore if the logstore of the arctic table is enabled. <br> 'auto' is recommended.                              |
| arctic.emit.auto-write-to-logstore.watermark-gap     | none |Duration | Only enabled when 'arctic.emit.mode'='auto', if the watermark of the arctic writers is greater than the current system timestamp subtracts the specific value, writers will also write data into the logstore.  Default: the logstore writer is enabled immediately once the job is launched. <br>This value must be greater than 0.                            |

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by YesOrNo828**
## Why are the changes needed?

Fix #389 
Automatic toggle switch to enable double write

## Brief change log

  - *Add the 'arctic.emit.mode' configuration with the enumeration: auto*
  - *Add the 'arctic.emit.auto-write-to-logstore.watermark-gap' configuration*
  - *The flink 1.12 1.14 1.15 versions are affected.*
  - *Add the documentation in the flink dml chapter*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (yes / no) yes
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented) docs

### CODE REVIEW

**Review by zhoujinsong — COMMENTED** [review]
@YesOrNo828  I left some comments. PTAL

**Review by zhoujinsong — APPROVED** [review]
LGTM.

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/descriptors/ArcticValidator.java [hunk 1]
```diff
@@ -136,27 +138,55 @@ public class ArcticValidator extends ConnectorDescriptorValidator {
...
+              " is enabled.\n" +
+              "'auto' is recommended.");
+
+
+  public static final ConfigOption<Duration> AUTO_EMIT_LOGSTORE_WATERMARK_GAP =
+      ConfigOptions.key("arctic.emit.auto-write-to-logstore.watermark-gap")
+          .durationType()
+          .noDefaultValue()
+          .withDescription("Only enabled when 'arctic.emit.mode'='auto', if the watermark of the arctic writers" +
+              " is greater than the current system timestamp subtracts the specific value, writers will also write" +
```
Comment (zstraw) [inline flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/descriptors/ArcticValidator.java:160]: Does the specific value mean the value of "arctic.emit.auto-write-to-logstore.watermark-gap"?

Comment (YesOrNo828) [reply flink/v1.12/flink/src/main/java/com/netease/arctic/flink/table/descriptors/ArcticValidator.java:160]: Yes.

File: flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/AutomaticLogWriter.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.runtime.tasks.StreamTask;
+import org.apache.flink.table.data.RowData;
+import org.apache.iceberg.Schema;
+
+import java.time.Duration;
+import java.util.Properties;
+
+public class AutomaticLogWriter extends ArcticLogWriter {
```
Comment (zstraw) [inline flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/AutomaticLogWriter.java:39]: Could you add some annotations to describe the overall scenario for this automatic writer? Such as only writing the latest data(in the gap of AUTO_EMIT_LOGSTORE_WATERMARK_GAP from now) into log-store.

**Pull request merged**